fvtest/porttest/si.cpp

```diff
@@ -2362,6 +2368,289 @@ TEST(PortSysinfoTest, sysinfo_test_os_kernel_info)
+#else /* defined(LINUX) */
+	if (0 != omrsysinfo_cgroup_are_subsystems_available(0x7)) {
+		outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_cgroup_are_subsystems_available returned nonzero on nonlinux\n");
+	}
+#endif /* defined(LINUX) */
+
+	reportTestExit(OMRPORTLIB, testName);
+	return;
+}
+
+/**
+ * Test omrsysinfo_cgroup_get_enabled_subsystems, omrsysinfo_cgroup_enable_subsystems,
+ * and omrsysinfo_cgroup_are_subsystems_enabled.
+ */
+TEST_F(CgroupTest, sysinfo_cgroup_enable)
+{
+	OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+	const char *testName = "sysinfo_cgroup_enable";
+
+	reportTestEntry(OMRPORTLIB, testName);
+
+#if defined(LINUX)
+	uint64_t enabled = 0;
+	/* Clear enabled subsystems. */
+	ASSERT_EQ(omrsysinfo_cgroup_enable_subsystems(0), (uint64_t)0);
+	ASSERT_EQ(omrsysinfo_cgroup_get_enabled_subsystems(), (uint64_t)0);
+	ASSERT_EQ(
+			omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_CPU | OMR_CGROUP_SUBSYSTEM_MEMORY | OMR_CGROUP_SUBSYSTEM_CPUSET),
+			(uint64_t)0);
+
+	enabled = CgroupTest::available & OMR_CGROUP_SUBSYSTEM_CPU;
+	EXPECT_EQ(omrsysinfo_cgroup_enable_subsystems(OMR_CGROUP_SUBSYSTEM_CPU), enabled);
+	EXPECT_EQ(omrsysinfo_cgroup_get_enabled_subsystems(), enabled);
+	EXPECT_EQ(omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_CPU), enabled);
+	EXPECT_NE(omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_MEMORY), OMR_CGROUP_SUBSYSTEM_MEMORY);
+	EXPECT_EQ(
+			omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_CPU | OMR_CGROUP_SUBSYSTEM_CPUSET),
+			enabled);
+
+	/* This disables cpu subsystem. */
+	enabled = CgroupTest::available & (OMR_CGROUP_SUBSYSTEM_MEMORY | OMR_CGROUP_SUBSYSTEM_CPUSET);
+	EXPECT_EQ(
+			omrsysinfo_cgroup_enable_subsystems(OMR_CGROUP_SUBSYSTEM_MEMORY | OMR_CGROUP_SUBSYSTEM_CPUSET),
+			enabled);
+	EXPECT_EQ(
+			omrsysinfo_cgroup_get_enabled_subsystems(),
+			enabled);
+	EXPECT_EQ(omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_MEMORY), CgroupTest::available & OMR_CGROUP_SUBSYSTEM_MEMORY);
+	EXPECT_EQ(
+			omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_CPU | OMR_CGROUP_SUBSYSTEM_MEMORY | OMR_CGROUP_SUBSYSTEM_CPUSET),
+			enabled);
+
+	/* This disables cpu and memory subsystems. */
+	enabled = CgroupTest::available & OMR_CGROUP_SUBSYSTEM_CPUSET;
+	EXPECT_EQ(omrsysinfo_cgroup_enable_subsystems(OMR_CGROUP_SUBSYSTEM_CPUSET), enabled);
+	EXPECT_EQ(omrsysinfo_cgroup_get_enabled_subsystems(), enabled);
+	EXPECT_EQ(omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_CPUSET), enabled);
+	EXPECT_NE(omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_MEMORY), OMR_CGROUP_SUBSYSTEM_MEMORY);
+	EXPECT_EQ(
+			omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_CPU | OMR_CGROUP_SUBSYSTEM_CPUSET),
+			enabled);
+#else /* defined(LINUX) */
+	ASSERT_EQ(omrsysinfo_cgroup_get_enabled_subsystems(), (uint64_t)0);
+	ASSERT_EQ(omrsysinfo_cgroup_enable_subsystems(0x7), (uint64_t)0);
```

Comment:
`OMR_CGROUP_SUBSYSTEM_ALL` can be used. It is available on all platforms.
```suggestion
	ASSERT_EQ(omrsysinfo_cgroup_enable_subsystems(OMR_CGROUP_SUBSYSTEM_ALL), (uint64_t)0);
```